src/benchmarklib/benchmark_table_encoder.cpp

```diff
@@ -130,13 +132,30 @@ bool BenchmarkTableEncoder::encode(const std::string& table_name, const std::sha
   auto encoding_performed = false;
   const auto column_data_types = table->column_data_types();
 
-  for (const auto& chunk : table->chunks()) {
-    if (!is_chunk_encoding_spec_satisfied(chunk_encoding_spec, get_chunk_encoding_spec(*chunk))) {
-      ChunkEncoder::encode_chunk(chunk, column_data_types, chunk_encoding_spec);
-      encoding_performed = true;
-    }
+  // Encode chunks in parallel, using `hardware_concurrency + 1` worker
+  // Not using JobTasks here because we want parallelism even if the scheduler is disabled.
+  auto next_chunk = std::atomic_uint{0};
+  auto threads = std::vector<std::thread>{};
+
+  for (auto thread_id = 0u;
+       thread_id < std::min(static_cast<uint>(table->chunk_count()), std::thread::hardware_concurrency() + 1);
+       ++thread_id) {
+    threads.emplace_back([&] {
```

Comment:
Let me just unconstructively say that I am uncomfortable with every component potentially using their own threads. A few more places doing this and we will loose oversight... Can we somehow transparently pipe this into the Scheduler and use Workers if the Scheduler has some, or a ThreadPool if there are none?